src/Extensions/AzureBlobPayloads/Options/LargePayloadStorageOptions.cs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure.Core;
+
+// Intentionally no DataAnnotations to avoid extra package requirements in minimal hosts.
+namespace Microsoft.DurableTask;
+
+/// <summary>
+/// Options for externalized payload storage, used by SDKs to store large payloads out-of-band.
+/// Supports both connection string and identity-based authentication.
+///
+/// <example>
+/// Connection string authentication:
+/// <code>
+/// var options = new LargePayloadStorageOptions("DefaultEndpointsProtocol=https;AccountName=mystorageaccount;AccountKey=...");
+/// </code>
+///
+/// Identity-based authentication:
+/// <code>
+/// var options = new LargePayloadStorageOptions(
+///     new Uri("https://mystorageaccount.blob.core.windows.net"),
+///     new DefaultAzureCredential());
+/// </code>
+/// </example>
+/// </summary>
+public sealed class LargePayloadStorageOptions
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="LargePayloadStorageOptions"/> class.
+    /// Parameterless constructor required for options activation.
+    /// </summary>
+    public LargePayloadStorageOptions()
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="LargePayloadStorageOptions"/> class.
+    /// </summary>
+    /// <param name="connectionString">The Azure Storage connection string to the customer's storage account.</param>
+    public LargePayloadStorageOptions(string connectionString)
+    {
+        Check.NotNullOrEmpty(connectionString, nameof(connectionString));
+        this.ConnectionString = connectionString;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="LargePayloadStorageOptions"/> class.
+    /// </summary>
+    /// <param name="accountUri">The Azure Storage account URI.</param>
+    /// <param name="credential">The credential to use for authentication.</param>
+    public LargePayloadStorageOptions(Uri accountUri, TokenCredential credential)
+    {
+        Check.NotNull(accountUri, nameof(accountUri));
+        Check.NotNull(credential, nameof(credential));
+        this.AccountUri = accountUri;
+        this.Credential = credential;
+    }
+
+    /// <summary>
+    /// Gets or sets the threshold in bytes at which payloads are externalized. Default is 900_000 bytes.
+    /// </summary>
+    public int ExternalizeThresholdBytes { get; set; } = 900_000; // leave headroom below 1MB
```

Comment:
Should we enforce a max this can be? Cause a customer could set it >1MB and have their requests fail, right?